### PR TITLE
Added hide password option

### PR DESCRIPTION
**Pull request by shiv07tiwari**
Added hide password option with Improved Dialog UI.
OK button only enabled if password is not null. @Swati4star

### CODE REVIEW

**Comment by Swati4star** [comment]
@shiv07tiwari Post the screenshot of dialog

**Comment by shiv07tiwari** [comment]
img_20180626_170854

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -97,6 +97,10 @@
     <string name="feedback_text">Kindly write your previous feeback, or just drop a Hi! :)</string>
     <string name="snackbar_no_email_clients">There are no email clients installed.</string>
     <string name="feedback_chooser">"Send mail…"</string>
+    <string name="example_pass_123">Example: pass@123</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: Add strings to other locale strings.xml also. Let them be in English only.

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 2]
```diff
@@ -552,37 +560,55 @@ private void passwordProtectPDF() {
...
-                        if (StringUtils.isEmpty(input)) {
+                    public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
+
+                    @Override
+                    public void onTextChanged(CharSequence s, int start, int before, int count) {
+                        positiveAction.setEnabled(s.toString().trim().length() > 0);
+                    }
+
+                    @Override
+                    public void afterTextChanged(Editable s) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: Change `s` to `input`

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 3]
```diff
@@ -66,11 +69,16 @@
     private static final int INTENT_REQUEST_GET_IMAGES = 13;
     private static final int PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT = 1;
     private static int mImageCounter = 0;
+    private static final String LOG_TAG = "!";
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: It is not being used. You can remove this.

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 4]
```diff
@@ -552,37 +560,55 @@ private void passwordProtectPDF() {
...
-                            mPassword = input.toString();
+                            mPassword = s.toString();
                             onPasswordAdded();
                         }
                     }
-                })
-                .positiveText(android.R.string.ok)
-                .negativeText(android.R.string.cancel);
+                });
+        if(StringUtils.isNotEmpty(mPassword)){
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: checkstyle will fail here

**Pull request closed without merging**